packages/@uppy/provider-views/src/ProviderView/ProviderView.jsx

```diff
@@ -100,29 +102,31 @@ export default class ProviderView extends View {
    * @param  {string} id Folder id
    * @returns {Promise}   Folders/files in folder
    */
-  getFolder (id, name) {
-    return this.sharedHandler.loaderWrapper(
-      this.provider.list(id),
-      (res) => {
-        const folders = []
-        const files = []
-        let updatedDirectories
-
-        const state = this.plugin.getPluginState()
-        const index = state.directories.findIndex((dir) => id === dir.id)
-
-        if (index !== -1) {
-          updatedDirectories = state.directories.slice(0, index + 1)
-        } else {
-          updatedDirectories = state.directories.concat([{ id, title: name }])
-        }
+  async getFolder (id, name) {
+    this.setLoading(true)
```

Comment:
Can't we derive loading state from the data (if there is no data and no error, then it is loading)? I find `setLoading` a bit error prone. I think I initially misunderstood when suggesting to remove `loaderWrapper`, I thought we didn't need to plugin state anymore, but that seems to be required t show the loading state.

Reply:
i don't know how that would work. wouldn't it then show a loading screen before any files have been selected(or loaded)?

Reply:
Not sure how that would work. Which state would we set to null or undefined?

Reply:
We can leave this as is for now I think